libpimeval/src/pimResMgr.h

```diff
@@ -44,6 +45,14 @@ class pimRegion
   uint64_t getElemIdxBegin() const { return m_elemIdxBegin; }
   uint64_t getElemIdxEnd() const { return m_elemIdxEnd; }
   uint64_t getNumElemInRegion() const { return m_elemIdxEnd - m_elemIdxBegin; }
+  unsigned getNumColsPerElem() const { return m_numColsPerElem; }
+
+  std::pair<unsigned, unsigned> locateIthElemInRegion(unsigned i) const {
+    assert(i < getNumElemInRegion());
+    unsigned rowIdx = m_rowIdx; // only one row of elements per region
+    unsigned colIdx = m_colIdx + i * m_numColsPerElem;
```

Comment:
for vertical layout isn't it multiple rows?

Reply:
Each region contains one row of elements, no matter if it's V or H layout. You can think each element takes a rectangle area of bits, and here we need to get the location of upper left corner.